### ISSUE TITLE
Unable to verify `icestick\leds` on Windows 10, seems to be an issue with SCONS

### ISSUE THREAD

**Issue by Nejat** [FPGAwars/apio#254]
I was following this very thorough tutorial, linked below, using `APIO` and I ran into this unexpected exception, different than the one described in the tutorial. I originally used the version mentioned in the tutorial `0.6.7`, but when that failed with an unexpected exception, I tried the latest version of `APIO` and still had issues. Basically, `apio verify` fails for the `icestick\leds` example.

[Introduction to FPGA Part 2 - Getting Started with Yosys, IceStorm, and Apio](https://youtu.be/gtkQ84Euyww)

Below is a transcript of what I captured as I proceeded though the steps of installing `APIO` and it's drivers, generating the example and running the verification. None of the other steps `sim, etc.` succeed either.

I attempted to debug the `SCONS` issue, but did not get very far.

It seems to be some `python` incompatibly, so I dropped down to the version of `python` on the video `3.8.2` and it still had the same issue.

I don't know if `python` follows the typical semantic versioning, but generally speaking `3.10.x` should not have any breaking changes from `3.8.x`

Any help figuring this out would be appreciated, Thanks

```
SHELL:   PowerShell 7.2.0
PYTHON:  3.10.0
PIP PACAKGES:
  apio               0.7.6
  certifi            2021.10.8
  charset-normalizer 2.0.7
  click              8.0.3
  colorama           0.4.4
  idna               3.3
  pip                21.3.1
  PyJWT              2.3.0
  pyserial           3.5
  requests           2.26.0
  SCons              4.2.0
  semantic-version   2.8.5
  setuptools         59.1.1
  urllib3            1.26.7
  wheel              0.37.0
```
```
> apio install --all

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing dfu package:
Download toolchain-dfu-windows_amd64-2020.11.24.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'dfu' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing drivers package:
Warning: full platform does not match: windows_amd64
         Trying OS name: windows
Download tools-drivers-windows-1.1.0.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'drivers' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing ecp5 package:
Download toolchain-ecp5-windows_amd64-2019.12.12.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'ecp5' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing examples package:
Download apio-examples-0.0.30.zip
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'examples' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing fujprog package:
Download toolchain-fujprog-windows_amd64-2020.10.6.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'fujprog' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing gtkwave package:
Warning: full platform does not match: windows_amd64
         Trying OS name: windows
Download tool-gtkwave-windows-3.3.77.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'gtkwave' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing ice40 package:
Download toolchain-ice40-windows_amd64-2019.12.11.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'ice40' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing icesprog package:
Download toolchain-icesprog-windows_amd64-1.0.0.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'icesprog' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing iverilog package:
Download toolchain-iverilog-windows_amd64-1.1.1.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'iverilog' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing oss-cad-suite package:
Download tools-oss-cad-suite-windows_amd64-0.0.6.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'oss-cad-suite' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing scons package:
Download scons-3.0.1.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'scons' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing verilator package:
Download toolchain-verilator-windows_amd64-1.0.0.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'verilator' has been successfully installed!
(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing yosys package:
Download toolchain-yosys-windows_amd64-2019.12.11.tar.gz
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Package 'yosys' has been successfully installed!

> apio install drivers

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Installing drivers package:
Already installed. Version 1.1.0

> apio drivers --ftdi-enable

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Launch drivers configuration tool

   FTDI driver installation:
   Usage instructions

      1. Connect the FTDI FPGA board
      2. Select (Interface 0)
      3. Replace driver by "libusbK"
      4. Reconnect the board
      5. Check `apio system --lsftdi`

```

![image](https://user-images.githubusercontent.com/789809/142229437-80336954-652f-4bab-9264-ba9579a33665.png)

```

FTDI drivers configuration finished

> apio system --lsftdi

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
(DEBUG) Run Command: lsftdi
(DEBUG) System_base_dir: C:\Users\nejat\.apio\packages\tools-oss-cad-suite
(DEBUG) System bin dir: C:\Users\nejat\.apio\packages\tools-oss-cad-suite\bin
(DEBUG) Executable file: C:\Users\nejat\.apio\packages\tools-oss-cad-suite\bin\lsftdi.exe
Number of FTDI devices found: 1
Checking device: 0
Manufacturer: Lattice, Description: Lattice FTUSB Interface Cable

> apio examples -d icestick\leds

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
Creating icestick\leds directory ...
Example 'icestick\leds' has been successfully created!

> cd .\icestick\leds\

> apio verify

(DEBUG) Profile path: C:\Users\nejat\.apio\profile.json
(DEBUG) Home_dir: C:\Users\nejat\.apio
SCons import failed. Unable to find engine files in:Traceback (most recent call last):
C:\Users\nejat\.apio\packages\tool-scons\script\..\engine

C:\Users\nejat\.apio\packages\tool-scons\script\scons-local-3.0.1File "C:\Users\nejat\.apio\packages\tool-scons\script\scons", line 192, in <module>

C:\Users\nejat\.apio\packages\tool-scons\script\scons-local
import SCons.Scriptc:\users\nejat\appdata\local\programs\python\python310\lib\site-packages\scons-3.0.1

C:\Users\nejat\AppData\Local\Programs\Python\Python310\scons-3.0.1File "C:\Users\nejat\.apio\packages\tool-scons\script\..\engine\SCons\Script\__init__.py", line 80, in <module>

C:\Users\nejat\AppData\Local\Programs\Python\Python310\Lib\site-packages\scons-3.0.1import SCons.Action

c:\users\nejat\appdata\local\programs\python\python310\lib\site-packages\sconsFile "C:\Users\nejat\.apio\packages\tool-scons\script\..\engine\SCons\Action.py", line 113, in <module>

C:\Users\nejat\AppData\Local\Programs\Python\Python310\scons
import SCons.ErrorsC:\Users\nejat\AppData\Local\Programs\Python\Python310\Lib\site-packages\scons

File "C:\Users\nejat\.apio\packages\tool-scons\script\..\engine\SCons\Errors.py", line 34, in <module>
import SCons.Util
File "C:\Users\nejat\.apio\packages\tool-scons\script\..\engine\SCons\Util.py", line 49, in <module>
from collections import Iterable
ImportError: cannot import name 'Iterable' from 'collections' (C:\Users\nejat\AppData\Local\Programs\Python\Python310\lib\collections\__init__.py)
```

**Comment by Nejat** [comment]
dropping down to python 3.8.10 let me run verify without issues

however, before switching python versions changing line `49` in `tool-scons\engine\SCons\Util.py`

from

```python
from collections import Iterable
```

to

```python
try:
    from collections.abc import Iterable
except ImportError:
    from collections import Iterable
```

also got `apio verify` working for the `icestick\leds` example

_I did attempt to create a pull request to address the problem, but `tox` barked that it needed another version and I was not going to trace that too, nor was I going to create a pull request without testing, so I hope this note helps._

**Comment by Obijuan** [comment]
Thanks for the feedback 😀️. The problem is that apio is using an older version of scons. It has been fixed in the development version of apio, which uses the latest scons. But the development version is broken currently because we are updating the toolchains to a new ones (so, many things are currently broken)

I will keep this issue open for testing it with the development version. I will ask you in the future if you could test it again

**Comment by Nejat** [comment]
👍

**Comment by Obijuan** [comment]
This error has been fixed in the latest apio version (0.8.1).Please, try it and feel free to reopen this issue if it remains (or open a new one in case new bugs are detected)
Thanks for the feedback
